src/kernels/raw/finder/localKernelFinder.node.ts

```diff
@@ -95,29 +92,49 @@ export class LocalKernelFinder implements ILocalKernelFinder, IExtensionSingleAc
                     !this.wasPythonInstalledWhenFetchingControllers &&
                     this.extensionChecker.isPythonExtensionInstalled
                 ) {
-                    this.updateCache(undefined).then(noop, noop);
+                    this.updateCache().then(noop, noop);
                 }
             },
             this,
             this.disposables
         );
+        this.nonPythonKernelFinder.onDidChangeKernels(
+            () => {
+                this.updateCache().then(noop, noop);
+            },
+            this,
+            this.disposables
+        );
+        this.pythonKernelFinder.onDidChangeKernels(
```

Comment:
Push model all the way down,
i.e. we detect changes from the sub components instead of polling.